verifier/src/main/java/com/exasol/classlistextractor/verifier/ClassListServer.java

```diff
@@ -0,0 +1,184 @@
+package com.exasol.classlistextractor.verifier;
+
+import java.io.*;
+import java.net.*;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import com.exasol.errorreporting.ExaError;
+
+import lombok.Getter;
+
+/**
+ * This class receives the class list from the agent running in the UDFs via TCP.
+ */
+class ClassListServer implements AutoCloseable {
+    private final ClientHandler clientHandler;
+
+    /**
+     * Create a new instance of {@link ClassListServer}.
+     */
+    public ClassListServer() {
+        this.clientHandler = new ClientHandler();
+    }
+
+    /**
+     * Get the port the server listens on.
+     * 
+     * @return port number
+     */
+    public int getPort() {
+        return this.clientHandler.getPort();
+    }
+
+    @Override
+    public void close() {
+        this.clientHandler.stop();
+    }
+
+    /**
+     * Clear the captured class list.
+     */
+    public void clear() {
+        this.clientHandler.clear();
+    }
+
+    /**
+     * Get the received class list.
+     * <p>
+     * The class list is a union (with duplicate removal) of all revieved class lists since the last call to
+     * {@link #clear()}.
```

Comment:
```suggestion
     * The class list is a union (with duplicates removed) of all retrieved class lists since the last call to
     * {@link #clear()}.
```